### PR TITLE
S3440 FN: Add test cases for is expression

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
This PR is closing issue #6390 since it's not C#11 related, the test will remain marked as FN and another issue will be opened to address the 'is' and 'is not' expressions for this rule.

### CODE REVIEW

**Comment by cristian-ambrosini-sonarsource** [comment]
Issue #6447 opened for fixing FN with 'is' and 'is not' expressions

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs [hunk 1]
```diff
@@ -133,3 +133,22 @@ public static Uri ToWebsocketUri(this Uri uri)
...
+class Repro6447
+{
+    void MyMethod()
+    {
+        string x = null;
+
+        if (!(x is null)) // FN (is expression not supported yet)
+        {
+            x = null;
+        }
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs:147]: This is redundant, you have already added it to the base test cases.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Left one small comment.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**